src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpSystemProxy.cs

```diff
@@ -110,16 +111,29 @@ private static Uri GetUriFromString(string value)
                 return null;
             }
 
-            if (!value.Contains("://"))
+            int idx = value.IndexOf("http://");
+            if (idx >= 0)
             {
-                value = "http://" + value;
+                int endOfProxy = value.IndexOfAny(_proxyDelimiters, idx);
+                int proxyLength = (endOfProxy == -1) ? value.Length - idx : endOfProxy - idx;
+
+                if (Uri.TryCreate( value.Substring(idx, proxyLength) , UriKind.Absolute, out Uri uri))
+                {
+                    return uri;
+                }
             }
 
-            if (Uri.TryCreate(value, UriKind.Absolute, out Uri uri) &&
-                    (uri.Scheme == Uri.UriSchemeHttp || uri.Scheme == Uri.UriSchemeHttps))
+            idx = value.IndexOf("http=");
+            if (idx >= 0)
             {
-                // We only support http and https for now.
-                return uri;
+                idx += 5; // Skip "http=" so we can replace it with "http://"
+                int endOfProxy = value.IndexOfAny(_proxyDelimiters, idx);
+                int proxyLength = (endOfProxy == -1) ? value.Length - idx : endOfProxy - idx;
+
+                if (Uri.TryCreate( "http://" + value.Substring(idx, proxyLength) , UriKind.Absolute, out Uri uri))
+                {
+                    return uri;
+                }
```

Comment:
When this string contains an https, isn't that about using that proxy to access https addresses, rather than accessing the proxy over https?  We support the former, not the latter.  I would think this code needs to be checking for https addresses as well.  Looks like that could entail some minor refactoring.

Reply:
In fact I don't think we should make any changes to GetUriFromString.  This seems to be used in two places: parsing the proxy config (the case we care about), and processing the result from GetProxyForUrl.  The former usage is just *wrong*, because it's not a simple Url.  We should change the config parsing code to not use this function.

Reply:
@stephentoub Thanks for clarifying -- that makes more sense than what I was thinking. I'll make that fix in the next iteration.

@geoffkizer I can split this out into a separate function to make that distinction clearer. Something like `ParseProxyConfig`. Would that be better?